lib/siwe/message.rb

```diff
@@ -156,11 +151,59 @@ def self.from_json_string(str)
       )
     end
 
-    def validate(signature)
-      raise Siwe::ExpiredMessage if !@expiration_time.empty? && Time.now.utc > Time.parse(@expiration_time)
-      raise Siwe::NotValidMessage if !@not_before.empty? && Time.now.utc < Time.parse(@not_before)
+    def validate
+      # check domain
+      raise Siwe::InvalidDomain unless @domain.match %r{[^/?#]*} || @domain.empty?
+
+      # check address EIP-55
+      raise Siwe::InvalidAddress unless Eth::Address.new(@address).to_s.eql? @address
```

Comment:
This will fail if an downcase or upcase address will be used. Better use `.valid?` to make sure all valid cases are checked:

```suggestion
      raise Siwe::InvalidAddress unless Eth::Address.new(@address).valid?
```

https://q9f.github.io/eth.rb/Eth/Address.html#valid%3F-instance_method

Reply:
Notably, `Eth::Address.new` will already raise if the address is not valid, so I liked the previous logic more.